multihomes.dsc

```diff
@@ -0,0 +1,214 @@
+      - stop
+    - narrate "<[name]> home has been set to your current location" format:zc_home_text
+
+#----------------------------------#
+#add (playername):(homename) support :check:
+#----------------------------------#
+delhome_command:
+  type: command
+  name: delhome
+  description: used to delete a home you created with /sethome
+  usage: /delhome (name)
+  debug: false
+  tab completions:
+    1: <player.flag[homes].keys>
+  script:
+    - if <context.args.first.contains_any_text[:]||false>:
+      - if !<player.has_permission[zc.homes_admin]>:
+        - narrate "You do not have permission to perform this command" format:zc_home_text
+        - stop
+      - define player <server.match_offline_player[<context.args.first.before[:]||default>]||noone>
+      - define name <context.args.first.after[:]||default>
+      - define is_admin true
+      - if <[player]> == noone:
+        - narrate "<context.args.first.before[:]> Does not seem to be a player" format:zc_home_text
+        - stop
+    - else:
+      - define player <player>
+      - define name <context.args.first||default>
+      - define is_admin false
+    - if !<[player].flag[homes].keys.contains[<[name]>]>:
+      - narrate "You cannot delete a home <tern[<[is_admin]>].pass[<[player].name> does].fail[you do]> not have" format:zc_home_text
+      - stop
+    - flag <[player]> homes.<[name]>:!
+    - if <player.flag[mutehomes.home]||false>:
+      - goto smartskip
+    - narrate "<tern[<[is_admin]>].pass[<[player].name>'s <[name]>].fail[<[name]>]> has been removed" format:zc_home_text
+    - mark smartskip
+    - if <context.args.get[2]> == open_list:
+      - execute as_player "listhomes <context.args.get[3]> <context.args.get[4]>"
+
+#----------------------------------#
+#pages instead of 1 big ass list :check:
+#per world and on alphabetical order :check:
+#clicking on the home tp's you there :check:
+#potentially make a delete button :check:
+#----------------------------------#
+listhomes_command:
+  type: command
+  name: listhomes
+  description: lists all the homes you currently have
+  usage: /listhomes
+  debug: false
+  tab completions:
+    1: <player.flag[homes].values.parse[world.name].deduplicate>
+  script:
+    - if <context.args.first.contains_any_text[:]||false>:
+      - if !<player.has_permission[zc.homes_admin]>:
+        - narrate "You do not have permission to perform this" format:zc_home_text
+        - stop
+      - define is_admin true
+      - define player <server.match_offline_player[<context.args.first.before[:]>]||noone>
+      - if <[player]> == noone:
+        - narrate "<context.args.first.before[:]> does not seem to be a player on our server. spelled it correctly?" format:zc_home_text
+        - stop
+      - define cur_world <context.args.first.after[:]||<player.location.world.name>>
+    - else:
+      - define player <player>
+      - define cur_world <context.args.first||<[player].location.world.name>>
+      - define is_admin false
```

Comment:
A lot of this logic is repeated throughout home scripts, perhaps make it an injectable task